srdt_analysis/database_manager.py

```diff
@@ -33,12 +33,30 @@ async def get_connection(self):
         async with self.pool.acquire() as conn:
             yield conn
 
-    async def fetch_documents_by_source(self, source: str) -> DocumentsList:
+    async def fetch_documents_by_source(self, source: CollectionName) -> DocumentsList:
         async with self.get_connection() as conn:
-            result = await conn.fetch(
-                "SELECT * from public.documents WHERE source = $1 AND is_published = true AND is_available = true",
-                source,
-            )
+            if source == "contributions":
+                result = await conn.fetch(
+                    """
+                    SELECT * from public.documents 
+                    WHERE source = $1 
+                    AND is_published = true 
+                    AND is_available = true 
+                    AND document->>'content' IS NOT NULL 
+                    AND document->>'idcc' = '0000'
+                    """,
+                    source,
+                )
+            else:
+                result = await conn.fetch(
+                    """
+                    SELECT * from public.documents 
+                    WHERE source = $1 
+                    AND is_published = true 
+                    AND is_available = true
+                    """,
+                    source,
+                )
```

Comment:
Comme on aura probablement pas de masses d'exceptions comme ça je pense que c'est OK de faire un if. Tu peux peut-être simplifier encore un peu plus en faisant ça :

```suggestion
            query = """
                SELECT * from public.documents
                WHERE source = $1
                AND is_published = true
                AND is_available = true
            """
            if source == "contributions":
                query += " AND document->>'content' IS NOT NULL AND document->>'idcc' = '0000'"

            result = await conn.fetch(query, source)
```